.github/workflows/pull_request.yml

```diff
@@ -109,3 +109,94 @@ jobs:
       - uses: actions/checkout@v4
       - name: Check Spelling
         uses: crate-ci/typos@master
+
+  build:
+    name: build 
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Configure sccache
+        uses: actions/github-script@v6
+        with:
+          script: |
+            core.exportVariable('ACTIONS_CACHE_URL', process.env.ACTIONS_CACHE_URL || '');
+            core.exportVariable('ACTIONS_RUNTIME_TOKEN', process.env.ACTIONS_RUNTIME_TOKEN || '');
+
+      - name: Build xline image
+        run: |
+          docker run -q --rm -v $(pwd):/xline \
+            -e SCCACHE_GHA_ENABLED=on \
+            -e ACTIONS_CACHE_URL=${ACTIONS_CACHE_URL} \
+            -e ACTIONS_RUNTIME_TOKEN=${ACTIONS_RUNTIME_TOKEN} \
+            ghcr.io/xline-kv/build-env:latest \
+            cargo build --release --bin xline --bin benchmark --bin validation_lock_client
+
+          sudo apt-get install -y --force-yes expect
+          cd scripts
+          cp ../target/release/{xline,benchmark,validation_lock_client} .
+          ldd ./xline
+          ldd ./benchmark
+          cp ../fixtures/{private,public}.pem .
+          docker build . -t ghcr.io/xline-kv/xline:master
+
+          cd .. 
+          pwd
+          mkdir -p _output
+          docker save -o _output/xline.tar ghcr.io/xline-kv/xline:master
+          ls _output
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: xline
+          path: _output/xline.tar
```

Comment:
Obviously this can be refactored into a reusable workflow, But it will not be processed in this PR.

Reply:
Could you please to initiate an issue for that?

Reply:
Sure, created https://github.com/xline-kv/Xline/issues/700 to trace it.